script/auth.js

```diff
@@ -0,0 +1,165 @@
+        hideError(emailInput, "email-invalidate");
+    }
+    submitButtonInvalidate();
+}
+
+if (emailInput) {
+    emailInput.addEventListener("focusout", checkEmail);
+}
+
+// 비밀번호 검사
+function checkPassword() {
+    const passwordValue = passwordInput.value;
+
+    if (!passwordValue) {
+        showError(passwordInput, "password-empty");
+    } else if (passwordValue.length < 8) {
+        hideError(passwordInput, "password-empty");
+        showError(passwordInput, "password-invalidate");
+    } else {
+        isPasswordValidate = true;
+        hideError(passwordInput, "password-empty");
+        hideError(passwordInput, "password-invalidate");
+    }
+    submitButtonInvalidate();
+}
+
+if (passwordInput) {
+    passwordInput.addEventListener("focusout", checkPassword);
+}
+
+// 닉네임 검사
+function checkName() {
+    const nameValue = nameInput.value;
+    if (!nameValue) {
+        showError(nameInput, "name-empty");
+        isNameValidate = false;
+    } else {
+        isNameValidate = true;
+        hideError(nameInput, "name-empty");
+    }
+    submitButtonInvalidate();
+}
+
+if (nameInput) {
+    nameInput.addEventListener("focusout", checkName);
+}
+
+// 비밀번호 확인
+function checkPasswordConfirm() {
+    const passwordValue = passwordInput.value;
+    const passwordConfirmValue = passwordConfirmInput.value;
+
+    if (!passwordConfirmValue) {
+        showError(passwordConfirmInput, "password-confirm-empty");
+        isPasswordConfirmValidate = false;
+    } else if (passwordValue !== passwordConfirmValue) {
+        hideError(passwordConfirmInput, "password-confirm-empty");
+        showError(passwordConfirmInput, "password-confirm-invalidate");
+        isPasswordConfirmValidate = false;
+    } else {
+        isPasswordConfirmValidate = true;
+        hideError(passwordConfirmInput, "password-confirm-empty");
+        hideError(passwordConfirmInput, "password-confirm-invalidate");
+    }
+    submitButtonInvalidate();
+}
+
+if (passwordConfirmInput) {
+    passwordConfirmInput.addEventListener("focusout", checkPasswordConfirm);
+}
+
+// 조건 모두 충족 시 로그인, 회원가입 버튼 활성화
+function submitButtonInvalidate() {
+    if (signupForm) {
+        if (isEmailValidate && isPasswordValidate && isPasswordConfirmValidate && isNameValidate) {
+            signupButton.disabled = false;
+            window.location.href="/login.html";
+        } else {
+            signupButton.disabled = true;
+        }
+    } else if (loginForm) {
+        if (isEmailValidate && isPasswordValidate) {
+            loginButton.disabled = false;
+            window.location.href="/items.html";
+        } else {
+            loginButton.disabled = true;
+        }
+    }
+}
+
+// 처음에 비활성화 되도록
+submitButtonInvalidate();  
+
+// 비밀번호 표시/숨김 함수
+function passwordVisibility(event) {
+    const button = event.currentTarget; 
+    const passwordInput = button.previousElementSibling;
+    const visibilityIcon = button;
+    const isPasswordVisible = passwordInput.type;
```

Comment:
P1:
is로 시작하는 변수는 boolean값으로 하시는것이 좋습니다.

```suggestion
    const isPasswordVisible = passwordInput.type === 'text';
```